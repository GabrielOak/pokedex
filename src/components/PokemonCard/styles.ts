import styled from 'styled-components/native';
import colorType from '../../theme/colorTypes';

interface CardProps {
  type: string;
}

export const Card = styled.TouchableOpacity<CardProps>`
  flex: 0.5;
  background-color: ${({type}) => colorType[type] || 'lightblue'};
  border-radius: 20px;
  height: 150px;
  margin: 8px;
  padding: 13px;
`;

export const Title = styled.Text`
  color: white;
  font-size: 16px;
  font-weight: 700;
  text-transform: capitalize;
`;

export const PokemonID = styled.Text`
  color: gray;
  font-size: 16px;
  align-self: flex-end;
`;

export const InfoContainer = styled.View`
  flex: 1;
  flex-direction: row;
  justify-content: space-between;
`;

export const TypesContainer = styled.View`
  flex: 1;
  flex-direction: column;
  align-items: center;
`;

export const Image = styled.Image`
  width: 80px;
  height: 85px;
  align-self: flex-end;
`;
