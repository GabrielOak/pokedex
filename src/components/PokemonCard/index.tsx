import React from 'react';
import {PokemonType} from '../../contexts/PokemonContext';

import * as S from './styles';

interface PokemonCardProps {
  pokemon: PokemonType;
}

const PokemonCard: React.FC<PokemonCardProps> = ({pokemon}) => {
  return (
    <S.Card type={pokemon.types[0]}>
      <S.PokemonID>{`#0${pokemon.id}`}</S.PokemonID>
      <S.Title>{pokemon.name}</S.Title>
      <S.InfoContainer>
        <S.TypesContainer>
          {pokemon.types.map(type => (
            <S.TypeIndicator>
              <S.TypeIndicatorText key={`${pokemon.name}-${type}`}>
                {type}
              </S.TypeIndicatorText>
            </S.TypeIndicator>
          ))}
        </S.TypesContainer>
        <S.Image
          source={{
            uri: pokemon.image,
          }}
        />
      </S.InfoContainer>
    </S.Card>
  );
};
export default PokemonCard;
