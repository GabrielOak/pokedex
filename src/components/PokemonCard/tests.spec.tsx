import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {render} from '@testing-library/react-native';
import PokemonCard from '.';

const pokemonMock = {
  attack: 52,
  defense: 43,
  height: 0.6,
  hp: 39,
  id: '4',
  image: '',
  name: 'charmander',
  specialAttack: 60,
  specialDefence: 50,
  speed: 65,
  types: ['fire'],
  weight: 8.5,
};

describe('<PokemonCard />', () => {
  it('should render correcly', () => {
    const {getByText, toJSON} = render(
      <NavigationContainer>
        <PokemonCard pokemon={pokemonMock} />
      </NavigationContainer>,
    );

    const id = `#${String(pokemonMock.id).padStart(3, '0')}`;

    expect(getByText(pokemonMock.name)).toBeTruthy();
    expect(getByText(id)).toBeTruthy();
    pokemonMock.types.map(type => expect(getByText(type)).toBeTruthy());

    expect(toJSON()).toMatchSnapshot();
  });
});
