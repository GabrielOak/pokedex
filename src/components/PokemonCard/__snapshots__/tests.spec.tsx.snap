// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PokemonCard /> should render correcly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#e26768",
      "borderRadius": 20,
      "flexBasis": 0,
      "flexGrow": 0.5,
      "flexShrink": 1,
      "height": 150,
      "marginBottom": 8,
      "marginLeft": 8,
      "marginRight": 8,
      "marginTop": 8,
      "opacity": 1,
      "paddingBottom": 13,
      "paddingLeft": 13,
      "paddingRight": 13,
      "paddingTop": 13,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "alignSelf": "flex-end",
          "color": "gray",
          "fontSize": 16,
        },
      ]
    }
  >
    #004
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontSize": 16,
          "fontWeight": "700",
          "textTransform": "capitalize",
        },
      ]
    }
  >
    charmander
  </Text>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexDirection": "column",
            "flexGrow": 1,
            "flexShrink": 1,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(255, 255, 255, 0.2)",
              "borderRadius": 50,
              "marginRight": 5,
              "marginTop": 8,
              "minWidth": 70,
              "paddingBottom": 5,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 5,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 12,
                "fontWeight": "600",
                "textTransform": "capitalize",
              },
            ]
          }
        >
          fire
        </Text>
      </View>
    </View>
    <Image
      source={
        Object {
          "uri": "",
        }
      }
      style={
        Array [
          Object {
            "alignSelf": "flex-end",
            "height": 85,
            "width": 80,
          },
        ]
      }
    />
  </View>
</View>
`;
